app/views/courses/TeacherClassView.coffee

```diff
@@ -411,18 +414,31 @@ module.exports = class TeacherClassView extends RootView
         @courseInstances.add(courseInstance)
         return courseInstance.save()
         
+    # Automatically apply licenses to students if necessary
     .then =>
-      availablePrepaids = @prepaids.filter((prepaid) -> prepaid.status() is 'available')
+      availablePrepaids = @prepaids.filter((prepaid) -> prepaid.status() is 'available' and prepaid.includesCourse(courseID))
       unenrolledStudents = _(members)
         .map((userID) => @students.get(userID))
         .filter((user) => user.prepaidStatus() isnt 'enrolled')
         .value()
       totalSpotsAvailable = _.reduce(prepaid.openSpots() for prepaid in availablePrepaids, (val, total) -> val + total) or 0
-      if totalSpotsAvailable < _.size(unenrolledStudents)
+      
+      availableFullLicenses = @prepaids.filter((prepaid) -> prepaid.status() is 'available' and prepaid.get('type') is 'course')
+      numStudentsWithoutFullLicenses = _(members)
+        .map((userID) => @students.get(userID))
+        .filter((user) => user.prepaidType() isnt 'course' or user.prepaidStatus() isnt 'enrolled')
+        .size()
```

Comment:
Same, this seems to include students with starter licenses. Also, again `coursePrepaid` is always a prepaid of type course, so that shouldn't need to be checked.